File: python/challenges/unique_values.py
# countUniqueValues takes a sorted array of numbers and return the count of unique numbers
def countUniqueValues(numbers):
    # if len of numbers is 0 return 0
    if(len(numbers) == 0):
        return 0

    # create two pointers for the start of the list and next of it
    prev = 0
    next = 1

    # interate with the numbers
    while next < len(numbers):

        # if the values of two pointer are different
        if(numbers[prev] != numbers[next]):
            # increment prev pointer
            prev += 1
            # assign the value on position of next pointer to the position of prev pointer
            numbers[prev] = numbers[next]

        # increment next pointer
        next += 1

    # return the prev pointer plus 1
    return prev + 1

File: go/challenges/unique_values.go
package challenges

/*
CountUniqueValues takes a sorted slice of numbers
 and returns the count of unique numbers
*/
func CountUniqueValues(nums []int) int {
	// check if the len of the slice is 0 and return 0
	if len(nums) == 0 {
		return 0
	}

	// make a pointer for the start of the slice
	start := 0

	// interate with the slice with a pointer next to the start pointer while it's small than the length of the array
	for next := 1; next < len(nums); next++ {
		// check if the value on the pointer are diff
		if nums[start] != nums[next] {
			// move the start pointer one position
			start++
			// assign the value from the next pointer to start pointer
			nums[start] = nums[next]
		}
	}

	// return the start pointer + 1
	return start + 1
}
